beamer/configure/config.py

```diff
@@ -0,0 +1,44 @@
+import json
+import traceback
+from dataclasses import dataclass, field
+from pathlib import Path
+
+from apischema import ValidationError, deserialize, schema
+
+from beamer.agent.typing import URL
+
+
+@dataclass
+class ChainConfig:
+    name: str
+    rpc: URL
+    tokens: list[list[str]] = field(default_factory=list)
+
+
+@dataclass
+class Param:
+    key: str | int | None = field(default=None)
+    values: dict[str, str | int] = field(default_factory=dict)
+
+
+@dataclass
+class Command:
+    name: str
+    params: list[Param]
+
+
+@dataclass
+class Config:
+    deployment_dir: str
+    chains: dict[str, ChainConfig] = field(metadata=schema(min_items=1))
+    commands: list[Command] = field(metadata=schema(min_items=1))
+
+    @staticmethod
+    def from_file(config_file: Path) -> "Config":
+        with open(config_file) as f:
+            config = json.load(f)
+        try:
+            config = deserialize(Config, config)
+        except ValidationError:
+            traceback.print_exc()
+        return config
```

Comment:
It seems weird that we catch `ValidationError`, print the traceback and then just continue.
I assume this would result in `NameError` being raised, which would be misleading.

Personally, I'd just omit the whole `try...except` block and let the exception propagate, terminating the Python process (and our command) with an exit code. The traceback would be output automatically.

For comparison, in `scripts/deployment/config.py` we do this:
```
try:                                    
    config = deserialize(Config, config)
except ValidationError as exc:          
    raise ConfigValidationError from exc
```

Reply:
you are completely right